### PR TITLE
Add RHSMVersion to systems endpoint

### PR DESCRIPTION
**Pull request by AlesKas**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,10 +9,12 @@ require (
 	github.com/antihax/optional v1.0.0
 	github.com/aws/aws-sdk-go v1.30.9
 	github.com/cenkalti/backoff v2.2.1+incompatible // indirect
+	github.com/cpuguy83/go-md2man/v2 v2.0.0 // indirect
```
Comment (Josca) [inline go.mod:12]: I think we don't need to add any deps for this task.

File: manager/controllers/systems_test.go [hunk 2]
```diff
@@ -36,6 +36,7 @@ func TestSystemsDefault(t *testing.T) {
 	assert.Equal(t, "RHEL", output.Data[0].Attributes.OSName)
 	assert.Equal(t, "8", output.Data[0].Attributes.OSMajor)
 	assert.Equal(t, "1", output.Data[0].Attributes.OSMinor)
+	assert.Equal(t, "8.1", output.Data[0].Attributes.RHSMVersion)
```
Comment (Josca) [inline manager/controllers/systems_test.go:39]: Please, update also advisory_systems_test.go and system_detail_test.go. Similarly like os version is done.

**Review by Josca — CHANGES_REQUESTED** [review]
Thanks, just a few things to update.

File: dev/test_data.sql [hunk 3]
```diff
@@ -178,34 +178,34 @@ GRANT SELECT ON TABLE inventory.hosts TO cyndi_reader;
...
 ('00000000000000000000000000000003', '1', '00000000-0000-0000-0000-000000000003', '[{"key": "k1", "value": "val1", "namespace": "ns1"}, {"key": "k3", "value": "val4", "namespace": "ns1"}]',
- '2018-09-22 12:00:00-04', '2018-08-26 12:00:00-04', '2018-08-26 12:00:00-04', '{"sap_system": true, "operating_system": {"name": "RHEL", "major": 8, "minor": 1}}'),
+ '2018-09-22 12:00:00-04', '2018-08-26 12:00:00-04', '2018-08-26 12:00:00-04', '{"sap_system": true, "operating_system": {"name": "RHEL", "major": 8, "minor": 1}, "rhsm_version": {"version": "8.1"}}'),
 ('00000000000000000000000000000004', '1', '00000000-0000-0000-0000-000000000004', '[{"key": "k3", "value": "val4", "namespace": "ns1"}]',
- '2018-09-22 12:00:00-04', '2018-08-26 12:00:00-04', '2018-08-26 12:00:00-04', '{"sap_system": true, "operating_system": {"name": "RHEL", "major": 8, "minor": 2}}'),
+ '2018-09-22 12:00:00-04', '2018-08-26 12:00:00-04', '2018-08-26 12:00:00-04', '{"sap_system": true, "operating_system": {"name": "RHEL", "major": 8, "minor": 2}, "rhsm_version": {"version": "8.2"}}'),
```
Comment (Josca) [inline dev/test_data.sql:187]: could you please add one row where the "rhsm_version" will not be presented and add test for that? Just to be sure it will work like expected - API should return empty RHSMVersion string. Then we can merge it.

**Review by Josca — CHANGES_REQUESTED** [review]
Just one last ask.

File: dev/test_data.sql (hunk 3 above)
Comment (AlesKas) [reply dev/test_data.sql:187]: Sure

**Review by Josca — CHANGES_REQUESTED** [review]
Great, but now there are some conflicts probably with @MichaelMraka 's updates. So please resolve it.

**Review by Josca — APPROVED** [review]
well done, thanks
